test/functional/rpc_generateblock.py

```diff
@@ -5,33 +5,70 @@
 '''Test generateblock rpc.
 '''
 
+from decimal import Decimal
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
 from test_framework.util import (
     assert_equal,
     assert_raises_rpc_error,
 )
+from test_framework.messages import (
+    COIN,
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_TRUE,
+)
+from test_framework.util import (
+    satoshi_round,
+    hex_str_to_bytes,
+)
+from test_framework.wallet import MiniWallet
+
+def create_miniwallet_rawtx(utxo, scriptPubKey):
+        flat_fee = Decimal('0.001')
+        send_value = satoshi_round(utxo['value'] - flat_fee)
+
+        tx = CTransaction()
+        tx.vin = [CTxIn(COutPoint(int(utxo['txid'], 16), utxo['vout']))]
+        tx.vout = [CTxOut(int(send_value * COIN), scriptPubKey)]
+        tx.wit.vtxinwit = [CTxInWitness()]
+        tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        rawtx = tx.serialize().hex()
+        return rawtx
 
 
 class GenerateBlockTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        address = ADDRESS_BCRT1_P2WSH_OP_TRUE
+        scriptPubKey = hex_str_to_bytes(node.validateaddress(ADDRESS_BCRT1_P2WSH_OP_TRUE)['scriptPubKey'])
+        utxos = []
```

Comment:
I'm not convinced that we need to maintain our own utxo set. After some re-working (ie adding "a just-create-don't-send option for the MiniWallet.") I'm wondering if this can go away

Reply:
Yes it probably can go away. My only reasoning for maintaining this list was because I wasn't exactly sure how I could get MiniWallet access to the utxos created from `generateblock`, and even if it did, Miniwallet (in its current state) doesn't have the option to create a transaction not submitted to mempool.